Recipes page

sunshinemoua opened the pull request:
- Created Recipes page with dynamic rows for recent, favs, and collections
- Mobile responsive
- View all button routes to new page
- Clicking on card routes to specific recipe url

sunshinemoua committed:
Refactor AppSidebar component structure and update import paths

- Moved AppSidebar component to a new directory structure for better organization.
- Removed SidebarAction component as it is no longer needed.
- Updated import paths in layout.tsx to reflect the new location of AppSidebar.

sunshinemoua committed:
Update dependencies and enhance UI components

- Updated package-lock.json and package.json to reflect the new versions.
- Introduced a ScrollArea component for better scroll management.
- Created a Carousel component for displaying recipes and collections in a visually appealing manner.
- Implemented a Card component for consistent styling of recipe and collection items.
- Added global CSS styles to hide scrollbars for a cleaner UI.
- Created reusable RecipeRow component
- Created dummy data for recipes and collections to be used in the profile page.

sunshinemoua committed:
Remove unnecessary App.jsx file containing favorite recipes data

sunshinemoua committed:
Refactor recipe data structure and enhance UI components

- Changed the property name from 'coverImage' to 'image' in the Collection interface for consistency.
- Added new recipes and collections to the dummy data for a richer user experience.
- Created new pages for displaying favorite recipes, recent recipes, and recipe collections, utilizing the new RecipeCard component for better presentation.
- Updated the ProfilePage to dynamically render recipe rows, improving maintainability and readability.

sunshinemoua committed:
Refactor Profile and Recipes pages to streamline UI and enhance functionality

- Removed the RecipeRow component from the ProfilePage, simplifying the layout.
- Updated the RecipesPage to include a new RecipeRow component for displaying recent and favorite recipes, as well as recipe collections.
- Improved error handling for user authentication in the RecipesPage.
- Enhanced the overall structure and readability of both pages.

sunshinemoua committed:
Enhance Next.js configuration and streamline user authentication handling

- Added remote image patterns to the Next.js configuration for improved image handling.
- Simplified user authentication error handling in ProfilePage and RecipesPage by removing unnecessary destructuring of user data.
- Updated RecipeCard component to utilize Next.js Image component for better performance and responsiveness.

gurkirtheerey commented on app/data/dummy-recipes.ts:
```diff
@@ -0,0 +1,219 @@
+import { v4 as uuidv4 } from "uuid";
+
+export interface Recipe {
```
refer to recipeTypes

gurkirtheerey commented on app/data/dummy-recipes.ts:
```diff
@@ -0,0 +1,219 @@
+  description: string;
+}
+
+export interface Collection {
```
create new type, maybe we don't need this yet

gurkirtheerey commented on app/data/dummy-recipes.ts:
```diff
@@ -0,0 +1,219 @@
+  image: string;
+}
+
+export const recentRecipes: Recipe[] = [
```
dummy data, will be populatd when user can CRUD recipes

gurkirtheerey commented on app/profile/page.tsx:
```diff
@@ -2,22 +2,19 @@ import { createClient } from "@/lib/supabase/server";
 
```
working on next!

gurkirtheerey commented on app/recent-recipes/page.tsx:
```diff
@@ -0,0 +1,17 @@
+import { recentRecipes } from "@/app/data/dummy-recipes";
```
error handling for recipes??

sunshinemoua replied:
![Screenshot 2025-04-03 at 11 12 20 AM](https://github.com/user-attachments/assets/a8af6088-9983-48a4-bb3f-d824425b31d6)
Explore button will route user to the /discover which hasn't been created yet. WIP!

gurkirtheerey commented on app/recipe-collections/page.tsx:
```diff
@@ -0,0 +1,21 @@
+import { collections } from "@/app/data/dummy-recipes";
```
i see recent, collections and favs are all using the same <main></main>. maybe we can extract that into it's own component and pass the title and recipe array?

gurkirtheerey commented on app/recipes/page.tsx:
```diff
@@ -1,18 +1,99 @@
-import { redirect } from "next/navigation";
+import { RecipeCard } from "@/components/RecipeCard";
+
+interface RecipeRowProps {
```
move out

gurkirtheerey commented on package.json:
```diff
@@ -11,19 +11,22 @@
     "tailwind-merge": "^3.0.2",
-    "tw-animate-css": "^1.2.4"
+    "tw-animate-css": "^1.2.4",
+    "uuid": "^11.1.0"
```
remember to delete package after connecting to db

gurkirtheerey commented on package.json:
```diff
@@ -32,6 +35,7 @@
     "@types/node": "^20",
     "@types/react": "^19",
     "@types/react-dom": "^19",
+    "@types/uuid": "^10.0.0",
```
same here

gurkirtheerey commented on next.config.ts:
```diff
@@ -1,7 +1,16 @@
 import type { NextConfig } from "next";
```
for <Image />

gurkirtheerey commented on components/RecipeCard.tsx:
```diff
@@ -0,0 +1,48 @@
+import Link from "next/link";
+import Image from "next/image";
+
+interface RecipeCardProps {
```
move to types file

gurkirtheerey commented on components/RecipeCard.tsx:
```diff
@@ -0,0 +1,48 @@
+  className?: string;
+}
+
+export function RecipeCard({ item, type, className = "" }: RecipeCardProps) {
```
do we need extra className here?

sunshinemoua committed:
Refactor recipe data handling and enhance UI components

- Cleared out the dummy data for recent recipes and replaced it with an empty array for error handling card.
- Introduced a new file, recipe-rows.ts, for row data
- Updated the FavoriteRecipesPage and RecentRecipesPage to utilize the new RecipeListPage component for a cleaner layout.
- Created new reusable RecipeCard, RecipeGrid, RecipeListPage, and RecipeRow components for consistent UI.
- Enhanced the recipe types to include optional properties for better flexibility in data handling.

sunshinemoua commented:
Commented out components related to Recipe Collections. Will continue this after recent and favs folders are implemented.
